### PR TITLE
feat(mme): AMF unit test for registration flow

### PR DESCRIPTION
**Pull request by panyogesh**
Fix:
      1. Introducing client servicer layer in AMF.
      2. Basic registration call flow unit testcase.

Test:
      1. Unit Testing.
      2. Basic call flow testing using simulator.

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/core/oai/common/itti_free_defined_msg.c [hunk 1]
```diff
@@ -234,6 +234,20 @@ void itti_free_msg_content(MessageDef* const message_p) {
...
+      break;
+
+    case NGAP_NAS_DL_DATA_REQ:
+      bdestroy_wrapper(&(NGAP_NAS_DL_DATA_REQ(message_p).nas_msg));
+      break;
+
+    case NGAP_INITIAL_CONTEXT_SETUP_REQ:
+      bdestroy_wrapper(&(NGAP_INITIAL_CONTEXT_SETUP_REQ(message_p).nas_pdu));
+      break;
+
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/common/itti_free_defined_msg.c]: rebase as  https://github.com/magma/magma/pull/9816 has also landed.

Comment (panyogesh) [reply lte/gateway/c/core/oai/common/itti_free_defined_msg.c]: > rebase as #9816 has also landed.

I rebased. Will just execute P0 and P1 and update.

File: lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+  rc = send_proc_authentication_info_answer(imsi, ue_id, true);
+  EXPECT_TRUE(rc == RETURNok);
+
+  /* Send uplink nas message for auth response from UE */
+  rc = send_uplink_nas_message_ue_auth_response(
+      amf_app_desc_p, ue_id, plmn, ue_auth_response_hexbuf,
+      sizeof(ue_auth_response_hexbuf));
+  EXPECT_TRUE(rc == RETURNok);
+
+  /* Send uplinkg nas message for security mode complete response from UE */
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/test/amf/test_amf_procedures.cpp:224]: typo `uplinkg`

**Review by ulaskozat — APPROVED** [review]
LGTM

**Pull request merged**